### DISCUSSION TITLE
Github Code Challenges

### DISCUSSION THREAD

**Discussion by 201flaviosilva** [community/community#18027] [Other Feature Feedback, Questions, & Ideas]
Hey there,

It would be nice if GitHub had a section for community code challenges.
I imagine anyone can create a challenge and anyone can try to solve it too. It could even have a world leaderboard for those who solved the most challenges. On the user page, solved challenges and ranking can appear.

I believe that being able to create/solve challenges created by other developers on GitHub and solved challenges appearing on my GitHub profile would be awesome. It's perfect if the contribution activity chart shows the challenges. 
And so GitHub would look like an amazing ecosystem.

With this feature, it would be much easier to share the GitHub profile for businesses as it is all in one place.

(I know that there are already websites and apps specialized in doing this like [code wars](https://www.codewars.com/dashboard), and [hackerrank](https://www.hackerrank.com/), but GitHub also created [GitHub Learnig Lab](https://lab.github.com/) when there are already a lot of course sites and etc... and as I said, it would all be in one place :) )

It’s just an idea 💡😅

**Comment by aamazing-coder** [comment]
+1
